upa/scripts/test_challenge

```diff
@@ -0,0 +1,248 @@
+    upa submit-proofs \
+           --keyfile ${USER_KEYFILE} \
+           --proofs-file ${UNIVERSAL_PROOFS_4} \
+           --proof-ids-file submit-multi-4.pids.json \
+           --wait > submit-multi-4.tx.hash
+
+    # Extract the submissions
+    upa get-submission `cat submit-1.tx.hash` > submit-1.submission.json
+    upa get-submission `cat submit-multi-4.tx.hash` > submit-multi-4.submission.json
+    
+    multi_4_submissionid=`jq -r .submissionId submit-multi-4.submission.json`
+
+
+    # Allocate the fees for the submitted proofs.
+    KEYFILE_PASSWORD=${DUMMY_PW} upa aggregator allocate-aggregator-fee \
+           --keyfile ${KEYFILE} \
+           --wait
+
+    # Extract proof refs, cids and inputs for the on-chain submission of 4
+    for i in {0..3} ; do
+        upa compute proof-ref \
+               --submission-file submit-multi-4.submission.json \
+               --proof-idx $i \
+               --proof-ref-file submit-multi-4.ref-$i.json
+        jq ".[$i]" ${UNIVERSAL_PROOFS_4} > submit-multi-4.proof-$i.json
+    done
+
+     # Check that no proofs are verified yet
+    (! upa query is-verified \
+            --proof-file ${TEST_SINGLE_PROOF})
+    for i in {0..3} ; do
+        (! upa query is-verified \
+                --proof-file submit-multi-4.proof-$i.json \
+                --proof-ref-file submit-multi-4.ref-$i.json)
+    done
+    ! upa query is-submission-verified ${multi_4_submissionid}
+
+    # Check stats.  Expect 2 submissions, 0 verified.
+    upa query stats > stats.1
+    if ! [ "3"  == "`jq -r .nextSubmissionIdx stats.1`" ] ; then
+        echo "FAILURE: next_idx expected to be 3" #
+        exit 1
+    fi
+    if ! [ "1" == "`jq -r .nextSubmissionIdxToVerify stats.1`" ] ; then
+        echo "FAILURE: last_verified expected to be 1"
+        exit 1
+    fi
+    allocated_fee_1=`jq -r .allocatedFee stats.1`
+    if [ "0" == "${allocated_fee_1}" ] ; then
+        echo "FAILURE: allocated_fee expected to be non-zero"
+        exit 1
+    fi
+
+    # Submit (fake) aggregated proof (from the deployer) for ONLY
+    # the submission of 4, skipping the single proof.
+    jq '.[0:4]' submit-multi-4.pids.json > temp_agg.pids.json
+    # Flatten proofIds into one array.
+    jq -s 'add' temp_agg.pids.json > agg.pids.json
+
+    # Compute the dummy calldata using `compute-final-digest`
+    upa aggregator compute-final-digest \
+           --proof-ids-file agg.pids.json \
+           --calldata-file agg.proof.calldata
+
+    submit_aggregated_proof_flags="\
+           --calldata-file agg.proof.calldata \
+           --submission submit-multi-4.submission.json \
+    "
+
+    # Submit the aggregated proof
+    KEYFILE_PASSWORD=${DUMMY_PW} upa aggregator submit-aggregated-proof \
+           --keyfile ${KEYFILE} \
+           ${submit_aggregated_proof_flags} \
+           --wait > agg-proof.hash
+
+    # Create a dump of the aggregated tx.
+    upa dev trace `cat agg-proof.hash` --trace-file agg-proof.trace.json
```

Comment:
What's this for?

Reply:
Removed. 

It is in `test_upa` script which I based this one off of. But doesn't seem to be used.